### PR TITLE
Upload an image to a Book

### PR DESCRIPTION
**Pull request by KateVlaar**
Now when adding a new book or editing an old one, by clicking 'Upload Photo' user may select a photo from their gallery to upload. Each book can only map to one image at an old time so if you attempt to upload an image when one already exists, the old one will be deleted. Photos are stored in Fire Storage and their download URL's are stored within the book and updated with FireStore accordingly.

### CODE REVIEW

**Comment by KateVlaar** [comment]
image
image

File: app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java [hunk 1]
```diff
@@ -77,12 +103,22 @@ protected void onCreate(Bundle savedInstanceState)
...
 
-    public void takePhoto(View view)
+    /**
+     * Allows the user to upload a photo for the book from their gallery
+     * of images on their phone
+     * https://developer.android.com/training/camera/photobasics
+     *
+     * @param view
+     */
+    public void uploadPhoto(View view)
```
Comment (naviza) [inline app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java:113]: rude.

**Review by naviza — APPROVED** [review]
I tested it and it looks great! Although regarding what Paul said about US 08.01.01, are we only letting the users upload pictures for now? If yes, we should add the camera option as a potential feature in the future.

**Review by IsaakC — COMMENTED** [review]
When I scroll through the My Requests and Borrow lists (with all the books) the images seem to apply to the wrong books and also swap during scrolling

**Comment by huddy987** [comment]
> When I scroll through the My Requests and Borrow lists (with all the books) the images seem to apply to the wrong books and also swap during scrolling

This might be an issue with the view in my Reuqests and Borrows and not an issue with this commit

**Review by huddy987 — CHANGES_REQUESTED** [review]
There's a bug with this PR:

1. Create a book without a picture
2. Click edit on the book and add a picture
3. The picture does not show up

This is the perfect PR to add 2 interface tests:
1. Create a new book with a picture
2. Create a new book, then edit it and add a picture

That way if this breaks in the future we will know

**Comment by naviza** [comment]
> There's a bug with this PR:
> 
> 1. Create a book without a picture
> 2. Click edit on the book and add a picture
> 3. The picture does not show up
> 
> This is the perfect PR to add 2 interface tests:
> 
> 1. Create a new book with a picture
> 2. Create a new book, then edit it and add a picture
> 
> That way if this breaks in the future we will know

Maybe Add this as a separate JIRA ticket? I can work on top of this branch and implement that.

**Comment by KateVlaar** [comment]
> > There's a bug with this PR:
> > 
> > 1. Create a book without a picture
> > 2. Click edit on the book and add a picture
> > 3. The picture does not show up
> > 
> > This is the perfect PR to add 2 interface tests:
> > 
> > 1. Create a new book with a picture
> > 2. Create a new book, then edit it and add a picture
> > 
> > That way if this breaks in the future we will know
> 
> Maybe Add this as a separate JIRA ticket? I can work on top of this branch and implement that.

I actually can't seem to reproduce it but I am more than happy to make the tests. However I did find a new bug which I am working on: 1. Add a new book without picture 2. Book appears to have a picture in MyBooks

**Comment by KateVlaar** [comment]
Fixed bug where wrong pictures show up with books. Unit tests to come in a future PR

**Review by IsaakC — APPROVED** [review]

**Pull request merged**